src/vehicle.cpp

```diff
@@ -6466,3 +6475,24 @@ void vehicle::calc_mass_center( bool use_precalc ) const
         mass_center_no_precalc_dirty = false;
     }
 }
+
+bool vehicle::is_floating() const
```

Comment:
If you're worried about this being called too often and becoming slow, back it up with a cached variable, just like mass is cached.